### PR TITLE
IE support related changes

### PR DESCRIPTION
**Pull request by MoumitaM**
## Description of the change

- Added logic for camelcase rather that using a NPM package.
- Set a specific version(3.0.2) of the set-value package where Reflect feature is not being used.
- Used the polyfill builder to build a CDN and dynamically load the package if any of the below features are missing.
1. Array.find
2. Array.includes
3. Promise
4. String.endsWith
5. String.includes
6. String.startsWith
7. Object.entries

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/449)

### CODE REVIEW

File: analytics.js [hunk 1]
```diff
@@ -1130,6 +1130,48 @@ class Analytics {
...
   }
 
+  /**
+   * Call control pane to get client configs
+   *
+   * @param {*} writeKey
+   * @memberof Analytics
+   */
+  load(writeKey, serverUrl, options) {
+    logger.debug("inside load ");
```
[the remark this answers is not in the document]
Comment (MoumitaM) [reply analytics.js]: commented out the line

File: analytics.js [hunk 2]
```diff
@@ -1130,6 +1130,48 @@ class Analytics {
...
+      !Array.prototype.includes ||
+      !Promise ||
+      !Object.entries
+    ) {
+      if (!window.hasOwnProperty("polyfill")) {
+        ScriptLoader(
+          "polyfill",
+          "https://polyfill.io/v3/polyfill.min.js?features=Array.prototype.find%2CArray.prototype.includes%2CPromise%2CString.prototype.endsWith%2CString.prototype.includes%2CString.prototype.startsWith%2CObject.entries"
+        );
+      }
```
Comment (saikumarrs) [inline analytics.js:1158]: We should improve this:
- Build a map of polyfill.io name vs feature ("Array.prototype.find": Array.prototype.find)
- If possible, move this to a different file.

Comment (MoumitaM) [reply analytics.js:1158]: If possible, move this to a different file - you mean the whole logic or just polyfill.io name vs feature mapping?

File: analytics.js [hunk 3]
```diff
@@ -1130,6 +1130,48 @@ class Analytics {
...
+    if (
+      !String.prototype.endsWith ||
+      !String.prototype.startsWith ||
+      !String.prototype.includes ||
+      !Array.prototype.find ||
+      !Array.prototype.includes ||
+      !Promise ||
+      !Object.entries
+    ) {
+      if (!window.hasOwnProperty("polyfill")) {
```
Comment (saikumarrs) [inline analytics.js:1153]: Is this still relevant?

File: analytics.js [hunk 4]
```diff
@@ -1130,6 +1130,48 @@ class Analytics {
...
+        );
+      }
+      const self = this;
+      const interval = setInterval(function () {
+        // check if the polyfill is loaded
+        if (window.hasOwnProperty("polyfill")) {
+          clearInterval(interval);
+          self.loadAfterPolyfill(writeKey, serverUrl, options);
+        }
+      }, 100);
```
Comment (saikumarrs) [inline analytics.js:1166]: How did we decide on `100` ms?

File: integrations/Fullstory/camelcase.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+// util function to convert the input to string type
```
Comment (saikumarrs) [inline integrations/Fullstory/camelcase.js:1]: I wouldn't put this in FullStory as it is a generic utility.

File: package.json [hunk 6]
```diff
@@ -72,7 +71,7 @@
     "regenerator-runtime": "^0.13.7",
     "rudder-component-cookie": "0.0.1",
     "semver": "^7.1.3",
-    "set-value": "^4.1.0",
+    "set-value": "3.0.2",
```
Comment (saikumarrs) [inline package.json:74]: Is this change still required?

Comment (saikumarrs) [reply package.json:74]: I didn't see the changes in other places for not using the package. As we discussed, it is not a recommended fix.

File: integrations/Fullstory/camelcase.js (hunk 5 above)
Comment (MoumitaM) [reply integrations/Fullstory/camelcase.js:1]: got it

File: integrations/Fullstory/camelcase.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+function convertToString(input) {
```
Comment (saikumarrs) [inline integrations/Fullstory/camelcase.js:2]: Please add unit test cases for this module

Comment (MoumitaM) [reply integrations/Fullstory/camelcase.js:2]: Will add later

File: package.json (hunk 6 above)
Comment (MoumitaM) [reply package.json:74]: done

File: analytics.js (hunk 2 above)
Comment (saikumarrs) [reply analytics.js]: I meant only the mapping for now

File: analytics.js (hunk 3 above)
Comment (saikumarrs) [reply analytics.js]: If it is not relevant, why are we using the same in the periodic checking logic of the polyfills load?

File: integrations/Fullstory/camelcase.js (hunk 7 above)
Comment (saikumarrs) [reply integrations/Fullstory/camelcase.js]: It'd not be a good idea to use something that is not properly tested.

File: analytics.js (hunk 3 above)
Comment (MoumitaM) [reply analytics.js]: After loading the polyfill we need to clear the interval and proceed. The checking is to make sure the loading is done.

Comment (MoumitaM) [reply analytics.js]: If `!String.prototype.endsWith ||
      !String.prototype.startsWith ||
      !String.prototype.includes ||` these are not preset that means polyfill is not present.

File: integrations/Fullstory/camelcase.js (hunk 7 above)
Comment (MoumitaM) [reply integrations/Fullstory/camelcase.js]: added

**Review by saikumarrs — APPROVED** [review]
